client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.ts

```diff
@@ -111,6 +112,16 @@ export class AssignmentDetailComponent extends BaseComponent implements OnInit {
      */
     private _assignment: ViewAssignment;
 
+    public get candidates(): ViewUser[] {
```

Comment:
Please undo this. `candidates` is auto-generated and the type `ViewAssignmentCandidate[]` is correct, so you can access the m2m model (analog to submitters). If you directly need users (It's just a shortcut...), use `this.candidatesAsUsers` (not `this.assignment.candidatesAsUsers`, it is not defined for the class `Assignment`).